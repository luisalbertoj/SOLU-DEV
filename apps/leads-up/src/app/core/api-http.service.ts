import { Injectable } from '@angular/core';
// Atento a esta línea, deberás hacer que empate con tu environment
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { environment } from '../../environments/environment';

import { Observable } from 'rxjs';

const API_URL = environment.apiRoute;

@Injectable({
  providedIn: 'root',
})
export class ApiHttpService {
  protected headers;

  protected apiUrl = API_URL;

  constructor(private http: HttpClient) {
    this.headers = {
      headers: new HttpHeaders({
        'Content-Type': 'application/json',
        Accept: 'application/json',
      }),
    };
  }

  protected get(url: string): Observable<unknown> {
    return this.http.get(this.apiUrl + url, this.headers);
  }

  protected post({ url, payload }: { url: string; payload: unknown }) {
    return this.http.post('api/' + url, payload, this.headers);
  }
}
