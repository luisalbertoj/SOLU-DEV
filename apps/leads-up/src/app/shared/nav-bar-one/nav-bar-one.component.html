<p-menubar [model]="items">
  <ng-template pTemplate="start">
    <img
      [src]="logo"
      [height]="height ?? '40'"
      alt="Logo compaÃ±ia"
      class="mr-2"
    />
  </ng-template>
  <ng-template pTemplate="end">
        
    </ng-template>
</p-menubar>
