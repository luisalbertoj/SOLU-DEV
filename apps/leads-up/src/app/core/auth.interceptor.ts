import {
  HttpEvent,
  HttpHandler,
  HttpInterceptor,
  HttpRequest,
} from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable()
export class AuthInterceptor implements HttpInterceptor {
  intercept(
    req: HttpRequest<unknown>,
    next: HttpHandler
  ): Observable<HttpEvent<unknown>> {
    const token = localStorage.getItem('auth_token');

    if (!token) {
      return next.handle(req);
    }

    const request = req.clone({
      headers: req.headers.set('Authorization', `Bearer ${token}`),
    });

    return next.handle(request);
  }
}
