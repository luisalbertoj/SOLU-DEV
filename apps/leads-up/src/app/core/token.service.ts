import { Injectable } from '@angular/core';

@Injectable()
export class TokenProvider {
  load() {
    return new Promise((resolve) => {
      resolve(true);
    });
  }
}
