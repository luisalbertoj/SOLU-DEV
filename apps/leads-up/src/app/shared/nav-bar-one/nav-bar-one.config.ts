import { Directive, Input } from '@angular/core';
import { MenuItem } from 'primeng/api';

@Directive()
export class NabBarOneConfig {
  /**
   * Items Of menu
   */
  @Input() items: MenuItem[] = [];
  /**
   * url of logo
   */
  @Input() logo?: string;
  /**
   * height of image logo
   */
  @Input() height?: string;
}
