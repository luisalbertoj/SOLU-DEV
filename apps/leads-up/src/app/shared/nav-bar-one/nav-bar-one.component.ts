import { CommonModule } from '@angular/common';
import { Component } from '@angular/core';
import { MenubarModule } from 'primeng/menubar';
import { NabBarOneConfig } from './nav-bar-one.config';

@Component({
  selector: 'solu-dev-nav-bar-one',
  standalone: true,
  imports: [CommonModule, MenubarModule],
  templateUrl: './nav-bar-one.component.html',
  styleUrls: ['./nav-bar-one.component.css'],
})
export class NavBarOneComponent extends NabBarOneConfig {
  constructor() {
    super();
  }
}
