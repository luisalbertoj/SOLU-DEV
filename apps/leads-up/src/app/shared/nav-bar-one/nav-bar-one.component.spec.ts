import { ComponentFixture, TestBed } from '@angular/core/testing';

import { NavBarOneComponent } from './nav-bar-one.component';

describe('NavBarOneComponent', () => {
  let component: NavBarOneComponent;
  let fixture: ComponentFixture<NavBarOneComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [NavBarOneComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(NavBarOneComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
