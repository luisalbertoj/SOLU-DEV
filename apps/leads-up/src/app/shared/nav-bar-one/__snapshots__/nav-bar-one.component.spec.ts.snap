// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarOneComponent this should be equal to the snapshot 1`] = `
"<p-menubar class=\\"p-element\\" ng-reflect-model=\\"\\"><div ng-reflect-ng-class=\\"[object Object]\\" class=\\"p-menubar p-component\\"><div class=\\"p-menubar-start\\"><img alt=\\"Logo compaÃ±ia\\" class=\\"mr-2\\" src=\\"\\" height=\\"40\\"><!--bindings={
  \\"ng-reflect-ng-template-outlet\\": \\"[object Object]\\"
}--></div><!--bindings={
  \\"ng-reflect-ng-if\\": \\"[object Object]\\"
}--><a tabindex=\\"0\\" class=\\"p-menubar-button\\"><i class=\\"pi pi-bars\\"></i></a><p-menubarsub root=\\"root\\" class=\\"p-element\\" ng-reflect-root=\\"root\\" ng-reflect-item=\\"\\" ng-reflect-base-z-index=\\"0\\" ng-reflect-auto-z-index=\\"true\\"><ul ng-reflect-ng-class=\\"[object Object]\\" role=\\"menubar\\" class=\\"p-menubar-root-list\\"><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"\\"
}--></ul></p-menubarsub><div class=\\"p-menubar-end\\"><!--bindings={
  \\"ng-reflect-ng-template-outlet\\": \\"[object Object]\\"
}--></div><!--bindings={
  \\"ng-reflect-ng-if-else\\": \\"[object Object]\\"
}--><!--container--></div></p-menubar>"
`;
